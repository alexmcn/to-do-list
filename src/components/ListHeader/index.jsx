import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import bgImage from './bg-image.png';

const useStyles = makeStyles({
  root: {
    padding: '100px 20px 20px',
    color: 'white',
    backgroundImage: `url(${bgImage})`,
    backgroundSize: 'cover',
    backgroundPosition: 'bottom',
  },
});

const ListHeader = ({ title, subtitle }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography gutterBottom variant="h3" component="h2" color="inherit">
        {title}
      </Typography>
      <Typography gutterBottom color="inherit" component="p">
        {subtitle}
      </Typography>
    </div>
  );
};

ListHeader.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
};

export default ListHeader;
