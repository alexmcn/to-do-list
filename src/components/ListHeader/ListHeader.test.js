import React from 'react';
import renderer from 'react-test-renderer';

import ListHeader from '.';

describe('ListHeader', () => {
  it('has a title and subtitle', () => {
    const instance = renderer.create(
      <ListHeader title="Team To-Do List" subtitle="Monday 1 July" />
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
