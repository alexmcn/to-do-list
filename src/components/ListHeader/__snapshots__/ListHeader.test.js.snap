// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader has a title and subtitle 1`] = `
<div
  className="makeStyles-root-1"
>
  <h2
    className="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit MuiTypography-gutterBottom"
  >
    Team To-Do List
  </h2>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit MuiTypography-gutterBottom"
  >
    Monday 1 July
  </p>
</div>
`;
